src/components/TokenDetailAction.js

```diff
@@ -0,0 +1,438 @@
+    return { promise, message: `${hathorLib.helpers.prettyValue(amountValue)} ${this.props.token.symbol} minted!` };
+  }
+
+  /**
+   * Method executed after user clicks on mint button. Validates the form.
+   *
+   * @return {string} Error message, in case of form invalid. Nothing, otherwise.
+   */
+  mint = () => {
+    if (this.chooseAddress.current.checked === false && this.address.current.value === '') {
+      return 'Address is required when not selected automatically';
+    }
+  }
+
+  /**
+   * Shows/hides address field depending on the checkbox click
+   *
+   * @param {Object} e Event for the address checkbox input change
+   */
+  handleCheckboxAddress = (e) => {
+    const value = e.target.checked;
+    if (value) {
+      $(this.addressWrapper.current).hide(400);
+    } else {
+      $(this.addressWrapper.current).show(400);
+    }
+  }
+
+  render() {
+    const renderMintAddress = () => {
+      return (
+        <div className="d-flex flex-row align-items-center justify-content-start col-9">
+          <div className="d-flex flex-row align-items-center address-checkbox">
+            <div className="form-check">
+              <input className="form-check-input" type="checkbox" ref={this.chooseAddress} id="autoselectAddress" defaultChecked={true} onChange={this.handleCheckboxAddress} />
+              <label className="form-check-label" htmlFor="autoselectAddress">
+                Select address automatically
+              </label>
+            </div>
+          </div>
+          <div className="form-group col-8" ref={this.addressWrapper} style={{display: 'none'}}>
+            <label>Destination address</label>
+            <input ref={this.address} type="text" placeholder="Address" className="form-control" />
+          </div>
+        </div>
+      );
+    }
+
+    const renderForm = () => {
+      return (
+        <div>
+          <div className="row">
+            <div className="form-group col-3">
+              <label>Amount</label>
+              <input required type="number" ref={this.amount} step={hathorLib.helpers.prettyValue(1)} min={hathorLib.helpers.prettyValue(1)} placeholder={hathorLib.helpers.prettyValue(0)} className="form-control" />
+            </div>
+            {renderMintAddress()}
+          </div>
+          <div className="form-group d-flex flex-row align-items-center">
+            <div className="form-check">
+              <input className="form-check-input" type="checkbox" ref={this.createAnother} id="keepMint" defaultChecked={true} />
+              <label className="form-check-label" htmlFor="keepMint">
+                Create another mint output for you?
+              </label>
+            </div>
+          </div>
+        </div>
+      )
+    }
+
+    return <TokenAction renderForm={renderForm} title='Mint tokens' buttonName='Go' validateForm={this.mint} onPinSuccess={this.executeMint} {...this.props} />
+  }
+}
+
+class TokenMelt extends React.Component {
```

Comment:
Add description of what the component does. In this case, it is used to melt tokens, right?

Reply:
Done